### PR TITLE
Test on more Pythons

### PR DESCRIPTION
**Pull request by cclauss**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,20 +2,26 @@
...
     - uses: actions/checkout@v3
     - uses: actions/setup-python@v4
       with:
-        python-version: 3.x
+        python-version: ${{ matrix.python-version }}
     - run: pip install --upgrade pip
-    - run: pip install pytest ruff .
-    - run: ruff --format=github --ignore=E501 --target-version=py37 .
+    - if: ${{ matrix.python-version }} == "3.11"
+      run: pip install ruff && ruff --format=github --ignore=E501 --target-version=py37 .
```
Comment (ywangd) [inline .github/workflows/ci.yml:25]: Is ruff still necessary? It seems tests are executed with pytests below anyway?

Comment (cclauss) [reply .github/workflows/ci.yml:25]: Ruff is a linter and pytest runs the test suite.
Pytest does not run the ruff tests.
To prove this, add a ruff error and only the Python 3.11 run will fail.

Comment (ywangd) [reply .github/workflows/ci.yml:25]: Cool thanks for explanation. It's my ignorance for what Ruff does.

**Comment by cclauss** [comment]
I should have included my usual commit message:

[Ruff](https://beta.ruff.rs/) supports [over 500 lint rules](https://beta.ruff.rs/docs/rules) and can be used to replace [Flake8](https://pypi.org/project/flake8/) (plus dozens of plugins), [isort](https://pypi.org/project/isort/), [pydocstyle](https://pypi.org/project/pydocstyle/), [yesqa](https://github.com/asottile/yesqa), [eradicate](https://pypi.org/project/eradicate/), [pyupgrade](https://pypi.org/project/pyupgrade/), and [autoflake](https://pypi.org/project/autoflake/), all while executing (in Rust) tens or hundreds of times faster than any individual tool.

**Pull request merged**